xdsl/rewriter.py

```diff
@@ -55,12 +55,15 @@ def replace_op(
             else:
                 old_result.replace_by(new_result)
 
-        op_idx = block.get_operation_index(op)
-        block.erase_op(op_idx, safe_erase=safe_erase)
         if len(op.results) == 0:
-            block.insert_op(new_ops, op_idx)
-            return
-        block.insert_op(new_ops, op_idx, op.results[0].name)
+            block.insert_ops_after(new_ops, op)
```

Comment:
You can move that line in both branches outside of the `if`, then you should just have a single `if` and no `else`.

Reply:
yeah I think I even had that at some point, must have lost that change